server/src/main/java/org/opensearch/index/shard/RemoteStoreSegmentUploadNotifier.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.shard;
+
+import org.opensearch.common.inject.Inject;
+import org.opensearch.indices.replication.checkpoint.ReplicationCheckpoint;
+import org.opensearch.indices.replication.checkpoint.SegmentReplicationCheckpointPublisher;
+
+/**
+ * Hook to publish notification after primary uploads segments to the remote store.
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreSegmentUploadNotifier {
```

Comment:
Doesn't look like we need this class because its a passthrough to `segRepPublisher.publish(indexShard, checkpoint);
` can we inject SegmentReplicationCheckpointPublisher directly into RemoteStoreRefreshListener?

Reply:
Yeah, Its passthrough as of now, but it might be useful when we've multiple notifications (For e.x CCR can have another one).